amd_openvx_extensions/amd_rpp/source/tensor/Normalize.cpp

```diff
@@ -55,19 +58,27 @@ static vx_status VX_CALLBACK refreshNormalize(vx_node node, const vx_reference *
         STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[1], VX_TENSOR_BUFFER_HIP, &roi_tensor_ptr, sizeof(roi_tensor_ptr)));
         STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_BUFFER_HIP, &data->pDst, sizeof(data->pDst)));
         STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[3], VX_TENSOR_BUFFER_HIP, &roi_tensor_ptr_dst, sizeof(roi_tensor_ptr_dst)));
+        if ((numDims == 4) && (!data->pSrcDims)) {  // For NHWC/NCHW layouts, allocate pSrcDims pinned memory buffer to store ROI for all dims
```

Comment:
Why do we need to check if numDims == 4 here? The memory is allocated based on actual roiDims queried from tensor. Please remove the check

Reply:
This check is added becoz for numDims = 3 (audio usecase - NHW/NFT/NTF) layout, the srcRoi buffer is created as BS * 2, not as BS * roiDims * 2. That's why there are different conditions added for numDims 3 and 4

Reply:
For audio layouts, roiDims is calculated differently. Removed the checks here and added the checks to assign roiDims accordingly